tasks/post_to_database/post_to_database.py

```diff
@@ -0,0 +1,249 @@
+"""
+Name: post_to_database.py
+
+Description:  Pulls entries from a queue and posts them to a DB.
+"""
+import datetime
+import json
+from typing import Any, List, Dict, Optional
+
+# noinspection SpellCheckingInspection
+import fastjsonschema as fastjsonschema
+from cumulus_logger import CumulusLogger
+from sqlalchemy import text
+
+from orca_shared import shared_db
+from orca_shared.shared_recovery import RequestMethod, OrcaStatus
+
+LOGGER = CumulusLogger()
+with open("schemas/new_job_input.json", "r") as raw_schema:
+    new_job_validate = fastjsonschema.compile(json.loads(raw_schema.read()))
+with open("schemas/update_file_input.json", "r") as raw_schema:
+    update_file_validate = fastjsonschema.compile(json.loads(raw_schema.read()))
+
+
+def task(records: List[Dict[str, Any]], db_connect_info: Dict):
+    for record in records:
+        send_record_to_database(record, db_connect_info)
+
+
+def send_record_to_database(record: Dict[str, Any], db_connect_info: Dict) -> None:
+    """
+    Deconstructs a record to its components and calls send_values_to_database with the result.
+
+    Args:
+        record: Contains the following keys:
+            'body' (str): A json string representing a dict.
+                Contains key/value pairs of column names and values for those columns.
+            'messageAttributes' (dict): Contains the following keys:
+                'TableName' (str): The name of the table to target.
+                'RequestMethod' (str): 'post' or 'put', depending on if row should be created or updated respectively.
+        db_connect_info: See shared_db.py's get_configuration for further details.
+    """
+    values = json.loads(record['body'])
+    request_method = RequestMethod(record['messageAttributes']['RequestMethod'])
+    if request_method == RequestMethod.NEW_JOB:
+        new_job_validate(values)
+        create_status_for_job_and_files(values['job_id'],
+                                        values['granule_id'],
+                                        values['request_time'],
+                                        values['archive_destination'],
+                                        values['files'],
+                                        db_connect_info)
+    elif request_method == RequestMethod.UPDATE_FILE:
+        update_file_validate(values)
+        update_status_for_file(values['job_id'],
+                               values['granule_id'],
+                               values['filename'],
+                               values['last_update'],
+                               values.get('completion_time', None),
+                               values['status_id'],
+                               values.get('error_message', None),
+                               db_connect_info)
+    else:
```

Comment:
Add a `logger.fatal` or `critical` message before the raise so it is captured in logs.

Reply:
Added